jujugui/static/gui/src/app/store/env/bakery.js

```diff
@@ -438,8 +449,14 @@ YUI.add('juju-env-bakery', function(Y) {
        @return {undefined} Nothing.
        */
       _successfulDischarges: function (originalRequest, jsonMacaroon) {
-        var prefix = this.macaroonName + '=';
-        document.cookie = prefix + btoa(JSON.stringify(jsonMacaroon));
+        if (this.nonCookie) {
+          window.localStorage.setItem(this.macaroonName,
+            btoa(JSON.stringify(jsonMacaroon)));
+        } else {
+          var prefix = this.macaroonName + '=';
+          document.cookie = prefix + btoa(JSON.stringify(jsonMacaroon))
```

Comment:
I'd mirror the logic for cookies here as I mentioned above for localStorage being passed in.